opentelemetry-sdk/src/logs/log_emitter.rs

```diff
@@ -136,22 +135,25 @@ impl LoggerProvider {
     }
 }
 
+impl Drop for LoggerProvider {
```

Comment:
what is the reason for implementing Drop for the LoggerProvider, as opposed to the LoggerProviderInner?